First implementation of JSON streamer plugin.

langrind opened the pull request:
First implementation of JSON streamer plugin. You can send JSON objects to port 5006 and it will make the values available for plotting.

langrind committed:
First implementation of JSON streamer plugin. You can send
JSON objects to port 5006 and it will make the values available
for plotting.

azaparov commented on plugins/DataStreamJSON/datastream_json.cpp:
```diff
@@ -0,0 +1,191 @@
+            QNetworkDatagram datagram = _sock->receiveDatagram();
+            pushSingleCycle(datagram);
+        }
+        std::this_thread::sleep_for ( std::chrono::milliseconds(10) );
```
Can this be done without sleeping ? Using polled socket ?

azaparov reviewed (commented):
Looks great i suggest removing sleep from loop function and use poll so that response is fast and if needed it can handle faster than 100hz update as well.

langrind committed:
Don't call sleep_for(), instead use the Qt socket method
waitForReadyRead() per azaparov's suggestion.

Simplify the approach a great deal : just expect data
points and plot them. In my application, I needed
a layer between the data generator and PlotJuggler,
so I started adding that layer in this plugin. But
azaparov suggested keeping the layer separate, which
turns out to be much nicer in my application, and
makes this plug-in simpler

langrind commented:
> Looks great i suggest removing sleep from loop function and use poll so that response is fast and if needed it can handle faster than 100hz update as well.
Done, good point, thank you!
